packages/iov-tendermint-rpc/src/v0-20/responses.ts

```diff
@@ -76,6 +76,25 @@ export class Responses {
     return decodeStatus(response.result as RpcStatusResponse);
   }
 
+  public static decodeSubscriptionEvent(event: JsonRpcEvent): responses.SubscriptionEvent {
+    // tslint:disable-next-line:no-let
+    let height: number;
+
+    if (event.data.value.header) {
+      // block header
+      height = event.data.value.header.height;
+    } else if (event.data.value.TxResult) {
+      // transaction
+      height = event.data.value.TxResult.height;
+    } else {
+      throw new Error("Unknown rpc event:" + JSON.stringify(event));
+    }
+
+    return decodeSubscriptionEvent({
```

Comment:
Woah, you really throw away all the other info except for the height of the change? So you just say "something happened, now go query"?

I kind of feel we should add some more info to the events. What do you think?

This can be another PR, but I was wondering why you did this instead of a union type for example.

Reply:
No I don't. This is just not ready